File: chapter01_math/01_simple_plots/sine_wave.m
% Run from octave command window

clc;  % Clear command history
close all;  % Close all windows
clear all;  % Clear variables


%% ******************** First approach ******************
z = 0: .1 : 12;  % Array from 0 to 12 with steps of.1
k = sin(z);

figure(1)  % Create figure
plot(z, k)

%% ******************** Second approach ******************
clear; clc;

T = 2 * pi;  % Period (s)
tf = 3 * T;  % End time (s)
t = 0: T / 20: tf;  % Time array (s)
w = 2 * pi / T;  % frequency (rad/s)
A = 1;  % Max amplitude (m)
x = sin(w * t);  % Position (m)

figure(2)
plot(t, x)

ylabel('x (m)')
xlabel('t (s)')
axes_handler = gca;  % gca: "get current axes"
set(axes_handler, 'FontSize',12);
title('Sine wave plot', 'FontSize', 20)


%% ******************** Third approach ******************
figure_handler = figure(3);
plot(t, x)

ylabel('x (m)')
xlabel('t (s)')
title('Sine wave plot with custom ticks')

xlim([0 6*pi]);
xticks([0: pi: tf])
xticklabels({'0','\pi','2\pi','3\pi','4\pi','5\pi','6\pi'})

ax = gca;
set(ax,'FontSize',12);

print(figure_handler, '-dpng', 'img/sine_octave.png')

File: chapter01_math/01_simple_plots/sine_wave.py
import numpy as np
import matplotlib.pyplot as plt


time = np.arange(0, 10, 0.1)

# Alternatively, we could create an array from 0 to 10, with 10 elements
# time = np.linspace(0, 10, 10)

y = np.sin(time)

plt.plot(time, y)

plt.title('Sine wave')
plt.xlabel('Time (s)')
plt.ylabel('Amplitude')

plt.axhline(y=0, color='k',ls=':')
plt.axvline(x=0, color='k',ls=':')

# plt.show()  # <----- Uncomment this line to view the plot
